hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionMetrics.java

```diff
@@ -0,0 +1,329 @@
+          e.getValue().getValue());
+      recordBuilder.endRecord();
+    }
+  }
+
+  /**
+   * Unregister the metrics instance.
+   */
+  public void unRegister() {
+    DefaultMetricsSystem.instance().unregisterSource(METRICS_SOURCE_NAME);
+  }
+
+  public synchronized void setTrackedDecommissioningMaintenanceNodesTotal(
+            long numNodesTracked) {
+    trackedDecommissioningMaintenanceNodesTotal
+        .set(numNodesTracked);
+  }
+
+  public synchronized void setTrackedRecommissionNodesTotal(
+          long numNodesTrackedRecommissioned) {
+    trackedRecommissionNodesTotal.set(numNodesTrackedRecommissioned);
+  }
+
+  public synchronized void setTrackedPipelinesWaitingToCloseTotal(
+          long numTrackedPipelinesWaitToClose) {
+    trackedPipelinesWaitingToCloseTotal
+        .set(numTrackedPipelinesWaitToClose);
+  }
+
+  public synchronized void setTrackedContainersUnderReplicatedTotal(
+          long numTrackedUnderReplicated) {
+    trackedContainersUnderReplicatedTotal
+        .set(numTrackedUnderReplicated);
+  }
+
+  public synchronized void setTrackedContainersUnhealthyTotal(
+          long numTrackedUnhealthy) {
+    trackedContainersUnhealthyTotal
+        .set(numTrackedUnhealthy);
+  }
+
+  public synchronized void setTrackedContainersSufficientlyReplicatedTotal(
+          long numTrackedSufficientlyReplicated) {
+    trackedContainersSufficientlyReplicatedTotal
+        .set(numTrackedSufficientlyReplicated);
+  }
+
+  public synchronized long getTrackedDecommissioningMaintenanceNodesTotal() {
+    return trackedDecommissioningMaintenanceNodesTotal.value();
+  }
+
+  public synchronized long getTrackedRecommissionNodesTotal() {
+    return trackedRecommissionNodesTotal.value();
+  }
+
+  public synchronized long getTrackedPipelinesWaitingToCloseTotal() {
+    return trackedPipelinesWaitingToCloseTotal.value();
+  }
+
+  public synchronized long getTrackedContainersUnderReplicatedTotal() {
+    return trackedContainersUnderReplicatedTotal.value();
+  }
+
+  public synchronized long getTrackedContainersUnhealthyTotal() {
+    return trackedContainersUnhealthyTotal.value();
+  }
+
+  public synchronized long getTrackedContainersSufficientlyReplicatedTotal() {
+    return trackedContainersSufficientlyReplicatedTotal.value();
+  }
+
+  private TrackedWorkflowContainerState createContainerMetricsInfo(
+      String host,
+      MetricByHost metric) {
+    return new TrackedWorkflowContainerState(host,
+        Interns.info(metric.getMetricName(),
+            metric.getDescription()), 0L);
+  }
+
+  public synchronized void metricRecordOfContainerStateByHost(
+      Map<String, ContainerStateInWorkflow> containerStatesByHost) {
+    trackedWorkflowContainerMetricByHost.clear();
+    for (Map.Entry<String, ContainerStateInWorkflow> e :
+        containerStatesByHost.entrySet()) {
+      trackedWorkflowContainerMetricByHost
+          .computeIfAbsent(MetricByHost.SufficientlyReplicated
```

Comment:
I don't think we need all the computIfAbsent calls here - we just cleared the map so they are always going to be absent, so just put the new value.